src/main/java/com/crowdin/util/CrowdinFileUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.crowdin.util;
+
+import com.crowdin.client.sourcefiles.model.Directory;
+import com.crowdin.client.sourcefiles.model.File;
+import lombok.NonNull;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CrowdinFileUtil {
+
+    private CrowdinFileUtil() {}
+
+    public static Map<String, File> buildFilePaths(@NonNull List<File> files, @NonNull Map<Long, Directory> dirs) {
+        Map<String, File> filePaths = new HashMap<>();
+        for (File file : files) {
+            StringBuilder sb = new StringBuilder(file.getName());
+            Long parentDirId = file.getDirectoryId();
+            while (parentDirId != null) {
+                Directory parentDir = dirs.get(parentDirId);
```

Comment:
There is a chance that map will not contain needed directory (just because we are loading only 500 entries). Please add extra validation here

Reply:
Changed to downloading all directories and files